pipelines/build/common/build_base_file.groovy

```diff
@@ -776,11 +776,17 @@ class Builder implements Serializable {
     /*
     Call job to push artifacts to github. Usually it's only executed on a nightly build
     */
-    def publishBinary(IndividualBuildConfig config=null) {
+    def publishBinary(IndividualBuildConfig config=null, String jobResult, String jobUrl) {
         def timestamp = new Date().format('yyyy-MM-dd-HH-mm', TimeZone.getTimeZone('UTC'))
         def javaVersion=determineReleaseToolRepoVersion()
         def stageName = 'BETA publish'
         def releaseComment = 'BETA publish'
+        def releaseWarning = ''
+        if ( jobResult != "SUCCESS" && jobResult != "UNSTABLE" ) {
+            // Build was not successful, add warning and link to build job
+            releaseWarning = '<a href=' + jobUrl + '><span style="color:red;">WARNING: build result(<b>' + jobResult + '</b>)</span></a> : '
```

Comment:
> > Does this mean build job status set as unsuccessful due to test failures will also generate this warning link?
> 
> if not SUCCESS|UNSTABLE, yes

In which case I'd recommend changing the messages to "pipeline result (FAILED|ABORTED)" since "build result" is potentially misleading e.g.

```suggestion
            releaseWarning = '<a href=' + jobUrl + '><span style="color:red;">WARNING: pipeline status was <b>' + jobResult + '</b></span></a> : '
```

Reply:
yep agreed, done